core/src/main/java/com/linecorp/armeria/client/Http2ClientConnectionHandler.java

```diff
@@ -110,8 +108,7 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
 
     @Override
     protected boolean needsImmediateDisconnection() {
-        return clientFactory.isClosing() ||
```

Comment:
If we use `clientFactory.isClosing()`, the stream is closed right away without waiting for the graceful shutdown.

Reply:
Oh right. Thanks for checking.